rdm/0007-access-control.md

```diff
@@ -0,0 +1,222 @@
+- As an administrator, I want to predefine roles, so that I can easily keep QA and production accurately reflect each other??
+- As an administrator, I want to restrict access to certain records based on user features (e.g. registered user, member of a specific community, member of academia), so that I can implement my institutions policies.
+- As a user, I want to access my data even after I have left my university.
+- As an administrator, I want to open my repositories for other institutions and their IDMS (?? Institutional Data Management System).
+
+## Terminology
+
+- **Resources**: The term resource in the system represents anything you want to protect. Resources can both be at the object-level (e.g. access to a record) or of more general nature (e.g access to the administration interface).
+Actions: Resources can be protected by requiring the ability to perform one or more actions. 
+- **Subjects**: Access to resources can be granted to *subjects*, which include users, roles and system roles. System roles are roles that are programmatically assigned to users. For example:
+    - Authenticated user: any logged in user.
+    - Any user: both authenticated and anonymous users.
+    - Campus user: Assigned to users based on their IP address being in a specific range.
+
+
+## Detailed design
+
+Invenio implements a Role-Based Access Control system (RBAC), that can be used to protect and grant access to resources.
+
+### Records as resources
+
+The most complex access rights use cases in Invenio RDM are related to records. Note that the following access control can be different between records and deposit records. In addition, this access control model is not tight to Invenio RDM, so it can be applied to any data model in Invenio.
+
+**Record actions**
+
+All records have the following basic CRUD actions:
+- Create: Ability to create new records.
+- Read: Ability to read a record (including during searches).
+- Update: Ability to update a record metadata.
+- Delete: Ability to delete a record.
+- List: Ability to perform searches (note, for a record to appear in the search result it is subject to the above mentioned read action).
+
+Records linked with files (possible for all records) have the additional following actions.
+- Create files: Ability to create files associated with a record. 
+- Read files: Ability to read files associated with a record.
+- Update files: Ability to update files associated with a record.
+- Delete files: Ability to delete files associated with a record.
+- Administrate files: Ability to perform all possible actions on files associated with a record, including reading previous versions of files.
+
+Deposit records have the additional following actions:
+- Publish: Ability to publish a new record.
+- Edit: Ability to edit an already published record. This differs from the record update action.
```

Comment:
Do you mean an unpublished record? Editing a published record seems to be the same as Updating a record.

Reply:
Yup... typo :)